lib/vdsm/taskset.py

```diff
@@ -95,6 +95,9 @@ def cpulist_parse(cpu_range):
         if '-' in item:
             begin, end = item.split('-', 1)
             cpus.extend(range(int(begin), int(end) + 1))
+        elif '^' in item:
```

Comment:
The format for exclusion is `^NUMBER`, so maybe it makes more sense to use `item.startswith("^")` here. Then we can try converting the number with `int(item[1:])`.

Another thing is that almost every example you will find on those strings will have the CPUs listed in a sorted manner, but I don't think there's a strict requirement for that (e.g. `taskset -cp 2,0 MYPID` will work fine). That means, that if someone uses a string like `^26,25-27`, the algorithm won't work. It would be better to keep the exluded CPUs in a separate list and return `frozenset(cpus) - frozenset(excluded_cpus)`.

Reply:
We have tests for this function in `domaindescriptor_test.py`, please add some test cases to cover the new logic.

Reply:
Updated the code and the test.